pkg/api/api.go

```diff
@@ -445,3 +445,12 @@ const (
 	// UserCancel user cancelled compose up, we are stopping containers
 	UserCancel
 )
+
+// GetDefaultImageName computes the default image name for a service, used to tag built images
+func GetDefaultImageName(service types.ServiceConfig, projectName string) string {
+	imageName := service.Image
+	if imageName == "" {
+		imageName = projectName + "_" + service.Name
```

Comment:
@glours Should we make this `-` to align with other resources (compose-spec/compose-go#294)?

Reply:
Oh, I see this is actually not a change in this PR, as this logic existed perviously as `getImageName` and is just exported now, so don't need to address before merging this even if we want to change

Reply:
It's less important for the image name, but I'll add a commit to use the compose separator 👍

Reply:
I'll do it in a PR to fix this issue #9618